infra/template/s3.tf

```diff
@@ -0,0 +1,59 @@
+# This bucket holds the state for terraform
+
+# bucket for hosting
+resource "aws_s3_bucket" "wic_mt_tf_state" {
+  bucket = "wic-mt-tf-state"
+  force_destroy = true
+}
+# enable versioning on bucket
+resource "aws_s3_bucket_versioning" "wic_mt_tf_state" {
+  bucket = aws_s3_bucket.wic_mt_tf_state.id
+  versioning_configuration {
+    status = "Enabled"
+  }
+}
+
+# encrypt data
+resource "aws_s3_bucket_server_side_encryption_configuration" "wic_mt" {
+  bucket = aws_s3_bucket.wic_mt_tf_state.bucket
+  rule {
+    apply_server_side_encryption_by_default {
+      sse_algorithm = "AES256"
+      }
+    }
+  }
+
+# block public access
+resource "aws_s3_bucket_public_access_block" "wic_mt_tf_state" {
+  bucket = aws_s3_bucket.wic_mt_tf_state.id
+
+  block_public_acls       = true
+  block_public_policy     = true
+  ignore_public_acls      = true
+  restrict_public_buckets = true
+}
+
+
```

Comment:
Any thoughts on adding bucket owner enforcement?

Reply:
I definitely like that plan